solvers/sk_oracle.py

```diff
@@ -0,0 +1,62 @@
+import warnings
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from flashcd import Lasso as FlashLasso
+    from sklearn.linear_model import Lasso
+    from sklearn.exceptions import ConvergenceWarning
+
+
+class Solver(BaseSolver):
+    """
+    Finds the correct support during setup with Scikit-Learn CD solver,
+    then runs CD on the design matrix restricted to the identified support.
+    """
+
+    name = "sk_oracle"
+
+    requirements = ["scikit-learn"]
+
+    def identify_support(self, X, y, lmbd, fit_intercept):
+        n_samples = X.shape[0]
+        clf = FlashLasso(
+            alpha=lmbd / n_samples,
+            tol=1e-12,
+            max_iter=1,
+            max_epochs=50_000,
+            fit_intercept=fit_intercept,
+            warm_start=False,
+            verbose=False,
+        )
+        warnings.filterwarnings("ignore", category=ConvergenceWarning)
+
+        beta = clf.fit(X, y)
+        supp = beta != 0
+        return supp
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
+        self.fit_intercept = fit_intercept
+
+        n_samples = self.X.shape[0]
+        self.clf = Lasso(
+            alpha=self.lmbd / n_samples, fit_intercept=fit_intercept, tol=0
+        )
+        warnings.filterwarnings("ignore", category=ConvergenceWarning)
+
+        supp = self.identify_support(X, y, lmbd, fit_intercept)
+        self.X = self.X[:, supp]
```

Comment:
I think if you do that here it will break because you will return a w/beta which is too small

Reply:
or you need to change the way `get_result` is implemented currently